Project/Src/Gui/GutterMargin.cs

```diff
@@ -42,7 +42,7 @@ public GutterMargin(TextArea textArea) : base(textArea)
         public override Cursor Cursor => RightLeftCursor;
 
         public override int Width
-            => textArea.TextView.WideSpaceWidth*Math.Max(4, (int)Math.Log10(textArea.Document.TotalNumberOfLines) + 4);
+            => textArea.TextView.WideSpaceWidth*Math.Max(2, (int)Math.Log10(textArea.Document.TotalNumberOfLines) + 1);
```

Comment:
Then, why not just
```suggestion
            => textArea.TextView.WideSpaceWidth * (int)Math.Log10(textArea.Document.TotalNumberOfLines);
```
?

Reply:
you need to add 1 to log10() should maybe be 2 to have a margin), and min 2 is to have a margin also for no lines (could be 1, if no margin is expected after 0)

Reply:
```suggestion
            => textArea.TextView.WideSpaceWidth * ((int)Math.Log10(Math.Max(1, textArea.Document.TotalNumberOfLines)) + 2);
```

Reply:
Made an alternative with adding half a character margin.

![image](https://github.com/user-attachments/assets/7877a919-1af1-40d5-bfb8-7c20982906c7) ![image](https://github.com/user-attachments/assets/3926f0c7-f443-4b7d-a2d0-25bb693c6e9d)

Reply:
(The gutter alignment bug is back - heavier than ever.)

Reply:
> (The gutter alignment bug is back - heavier than ever, also in v5.2.1.)

Unfortunately more visible here with 1-3 length, no longer at 1000 lines.
I do not see a workaround, suggest it is handled separately

Reply:
> I do not see a workaround, suggest it is handled separately

Yes, that is why I put the comment in parentheses.